src/main/java/io/mosip/mimoto/controller/CredentialsController.java

```diff
@@ -0,0 +1,94 @@
+package io.mosip.mimoto.controller;
+
+import io.mosip.mimoto.core.http.ResponseWrapper;
+import io.mosip.mimoto.dto.ErrorDTO;
+import io.mosip.mimoto.dto.IssuerDTO;
+import io.mosip.mimoto.dto.idp.TokenResponseDTO;
+import io.mosip.mimoto.dto.mimoto.CredentialIssuerWellKnownResponse;
+import io.mosip.mimoto.dto.mimoto.CredentialsSupportedResponse;
+import io.mosip.mimoto.dto.mimoto.VCCredentialRequest;
+import io.mosip.mimoto.dto.mimoto.VCCredentialResponse;
+import io.mosip.mimoto.exception.ApiNotAccessibleException;
+import io.mosip.mimoto.exception.IdpException;
+import io.mosip.mimoto.service.CredentialService;
+import io.mosip.mimoto.service.IdpService;
+import io.mosip.mimoto.service.IssuersService;
+import io.mosip.mimoto.util.DateUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.InputStreamResource;
+import org.springframework.http.*;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.client.RestTemplate;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import static io.mosip.mimoto.exception.PlatformErrorMessages.API_NOT_ACCESSIBLE_EXCEPTION;
+import static io.mosip.mimoto.exception.PlatformErrorMessages.MIMOTO_PDF_SIGN_EXCEPTION;
+
+@RestController
+@RequestMapping(value="/credentials")
+public class CredentialsController {
+
+    private static final String ID = "mosip.mimoto.credentials";
+    private final Logger logger = LoggerFactory.getLogger(CredentialsController.class);
+
+    @Autowired
+    IssuersService issuersService;
+
+    @Autowired
+    CredentialService credentialService;
+
+    @Autowired
+    IdpService idpService;
+
+    @PostMapping("/download")
+    public ResponseEntity<?> downloadCredentialAsPDF(
+            @RequestParam Map<String, String> params) {
+
+        ResponseWrapper<Object> responseWrapper = new ResponseWrapper<>();
+        responseWrapper.setId(ID);
+        responseWrapper.setVersion("v1");
+        responseWrapper.setResponsetime(DateUtils.getRequestTimeString());
+        try{
+
+            String issuerId = params.get("issuer");
+            String credentialType = params.get("credential");
```

Comment:
should we rename `credential` param in request to `credentialType`?

Reply:
"Credential" itself implicitly means the type of credential, i took the naming from wellknown specs.